source/draw/gpu/GpuSurface.ooc

```diff
@@ -83,7 +83,13 @@ GpuSurface: abstract class extends Canvas {
 			temporary = this _context createImage(image as RasterImage)
 		else
 			Debug raise("Invalid image type in GpuSurface!")
-		map add("texture0", temporary)
+		if (temporary instanceOf?(GpuYuv420Semiplanar)) {
+			yuv := temporary as GpuYuv420Semiplanar
+			map add("texture0", yuv y)
+			map add("texture1", yuv uv)
+		}
+		else
+			map add("texture0", temporary)
```

Comment:
Oh yes, the ugly `else` format rule, how could I forget... We probably want it in magic, I will never get used to it.